smack-experimental/src/main/java/org/jivesoftware/smackx/mam/MamManager.java

```diff
@@ -216,10 +220,13 @@ public static synchronized MamManager getInstanceFor(XMPPConnection connection,
 
     private final ServiceDiscoveryManager serviceDiscoveryManager;
 
+    private final AdHocCommandManager adHocCommandManager;
+
     private MamManager(XMPPConnection connection, Jid archiveAddress) {
         super(connection);
         this.archiveAddress = archiveAddress;
         serviceDiscoveryManager = ServiceDiscoveryManager.getInstanceFor(connection);
+        this.adHocCommandManager = AdHocCommandManager.getAddHocCommandsManager(connection);
```

Comment:
Just like `serviceDiscoveryManager` in the line above, you do not need to qualify the field with `this.` here.

Reply:
May I ask the reason for not qualifying field with this? is it just for readability ?

Reply:
Increased readability by removing unnecessary noise in form of keywords are one argument. The other one is homogeneity with the existing code.

Reply:
Updated please review